.github/workflows/backport.yml

```diff
@@ -0,0 +1,52 @@
+name: Backport merged pull request
+
+on:
+  pull_request_target:
+    types: [closed]
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  backport:
+    runs-on: ubuntu-latest
+    name: Backport PR
+    # Only run if the PR was merged (not just closed)
+    if: github.event.pull_request.merged == true
```

Comment:
Do we need to limit this application to `main` branch only  to prevent triggering on already backported branches ? 

```
if: github.event.pull_request.base.ref == 'main' && github.event.pull_request.merged == true
```

Reply:
we are going to manually add labels. so its not fine if we choose backport a backport PR. for example 
- main PR --> apply label: "backport branch-0.3" --> merge
- 0.3 PR from merged changes in main PR --> conflict --> resolved conflict --> apply label: "backport branch-0.2" --> merge 
   - _This is useful because now we know that after conflict resolution this diff may work for 0.2 as well, so backport this to 0.2_
- 0.2 PR from merged changes in 0.3 PR
wdyt?